sdk/src/impl/position-impl.ts

```diff
@@ -246,20 +251,58 @@ export class PositionImpl implements Position {
     let txBuilder = new TransactionBuilder(this.ctx.provider.connection, this.ctx.provider.wallet);
 
     const accountExemption = await this.ctx.fetcher.getAccountRentExempt();
-    const { ataTokenAddresses, resolveAtaIxs } = await resolveAtaForMints(this.ctx, {
-      mints: [whirlpool.tokenMintA, whirlpool.tokenMintB],
-      accountExemption,
-      receiver: destinationWalletKey,
-      payer: ataPayerKey,
-    });
+    const walletTokenAccountsByMint = { ...ownerTokenAccountsRecord };
+
+    const tokenAAndB = [whirlpool.tokenMintA, whirlpool.tokenMintB];
+    const tokenAorBIsSol = tokenAAndB.some((mint) => TokenUtil.isNativeMint(mint));
+    const solInWalletTokenAccounts = !!walletTokenAccountsByMint[NATIVE_MINT.toBase58()];
+    const mintsToResolveAtasFor = tokenAAndB.filter(
+      // 1. Filter out native mints since we don't use ATAs for them
+      // 2. Filter out mints for which we already have token accounts
+      (mint) => !TokenUtil.isNativeMint(mint) && !walletTokenAccountsByMint[mint.toBase58()]
+    );
 
-    const tokenOwnerAccountA = ataTokenAddresses[whirlpool.tokenMintA.toBase58()];
-    const tokenOwnerAccountB = ataTokenAddresses[whirlpool.tokenMintB.toBase58()];
+    if (mintsToResolveAtasFor.length > 0) {
```

Comment:
Can call resolveAta with an empty/native-mint-only dictionary to reduce complexity here. If it's empty, it'll return empty and you'll append empty onto your map later.